docs(#8425): Adds Documentation for Vault TOTP secret engine

lordofthejars opened the pull request.

lordofthejars committed:
docs(#8425): Adds Documentation for Vault TOTP secret engine

sberyozkin commented on docs/src/main/asciidoc/vault.adoc:
```diff
@@ -492,6 +492,45 @@ You should see:
+
+The TOTP secrets engine generates time-based credentials according to the TOTP standard.
+
+Vault supports both the generator scenario (like Google Authenticator) and the provider scenario (like the Google.com sign in).
```
And replace `Vault` with a `Vault TOTP` link as well

sberyozkin commented on docs/src/main/asciidoc/vault.adoc:
```diff
@@ -492,6 +492,45 @@ You should see:
+Vault supports both the generator scenario (like Google Authenticator) and the provider scenario (like the Google.com sign in).
+You can read more about Vault TOTP at https://www.vaultproject.io/docs/secrets/totp/.
+
+The Vault extension also supports TOTP by integrating with TOTP secret engine by providing an `io.quarkus.vault.VaultTOTPSecretEngine` class.
```
`also` can be dropped: `The Vault extension integrates with the http://...[Vault TOTP] secret engine by providing...`

sberyozkin commented:
Hi @lordofthejars This looks informative, thanks, I've proposed few minor updates
CC @gsmet, Hi Guillaume, please verify if what I've proposed makes sense or if it is already ready to go, etc, thanks

sberyozkin commented on docs/src/main/asciidoc/vault.adoc:
```diff
@@ -492,6 +492,45 @@ You should see:
+[[totp]]
+== TOTP Secrets Engine
+
+The TOTP secrets engine generates time-based credentials according to the TOTP standard.
```
Hmm, I've lost a comment, proposal was to replace `The TOTP secrets engine` with a link, https://www.vaultproject.io/docs/secrets/totp/[VaultTOTP] and drop "You can read more about..."

lordofthejars commented:
@sberyozkin I think I have addressed all your comments. I have added the link in just one place as if not we were repeating the same link in three consecutive paragraphs and I think it is not necessary.

@gsmet When @sberyozkin is ready we can merge this so it can be shown in 1.4.0.CR1. Notice that the logic is already there.

sberyozkin reviewed (approved):
Thanks @lordofthejars, IMHO we can merge it now, I can do it later today unless Guillaume or Vincent will add something.
